Modules/_ctypes/_ctypes.c

```diff
@@ -1150,6 +1147,12 @@ size property/method, and the sequence protocol.
 
 */
 
+/*[clinic input]
+class _ctypes.PyCPointerType "PyObject *" "st->PyCPointerType_Type"
```

Comment:
FWIW, I've used the practice of defining a "clinic state" macro for getting the module state in generated clinic code. It has proven useful if you want to tweak stuff afterwards (smaller diffs, less churn). You can `git grep clinic_state **/*.c` for inspiration.

Reply:
IIUC, `*_impl()` should be passed a module state if the AC gets it in advance with some overhead.  Also, `_ctypes` would need a module state getter to be specified in each function clinic input, making a class input have a type name without any operator.

Reply:
Ah, `_PyType_GetModuleState(cls)` or `_PyType_GetModuleState(cls->tp_base)` can be applied to each class. I'll try.